torch/_dynamo/guards.py

```diff
@@ -549,13 +670,15 @@ def compile_check_fn(self, local_builder, global_builder):
             verbose_code_parts.append(f"___check_tensors_verbose({verbose_args})")
 
         code = " and ".join(unique(code_parts))
-
         closure_vars = collections.OrderedDict(
             [
                 ("___guarded_code", self),
                 ("___check_tensors", check_tensors_fn),
                 ("___check_tensors_verbose", check_tensors_verbose_fn),
                 ("tensor_check_names", tensor_check_names),
+                ("Eq", sympy.Eq),
+                ("Mod", sympy.Mod),
+                ("FloorDiv", FloorDiv),
```

Comment:
Uhhh do you really want to be running sympy code when evaluating the guards?

Reply:
Why not?

Reply:
Because... guards are hotpath, and by calling sympy.Eq you're uselessly pinging sympy's Eq implementation which will typecheck the symbols and only then decide that a plain equality is OK?

Reply:
Sure, we can rewrite it as `==` and such.

Reply:
you could also do operator.eq if infix operators are annoying

Reply:
I'm gonna just inline my own eq w/ a closure that rewrites it as `a == b`

Reply:
You sure you don't wanna do Mod and FloorDiv too? lol